static/app/utils/performance/contexts/pageAlert.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {createContext, Fragment, useContext, useState} from 'react';
+import {Theme} from '@emotion/react';
+
+import {Alert} from 'sentry/components/alert';
+
+type PageAlertType = keyof Theme['alert'];
+
+export type PageAlertOptions = {
+  message: React.ReactNode | undefined;
+  type: PageAlertType;
+};
+
+type PageAlertSetter = (message: React.ReactNode | undefined) => void;
+
+const pageErrorContext = createContext<{
+  setPageError: PageAlertSetter;
+  setPageInfo: PageAlertSetter;
+  setPageMuted: PageAlertSetter;
+  setPageSuccess: PageAlertSetter;
+  setPageWarning: PageAlertSetter;
+  pageAlert?: PageAlertOptions;
+}>({
+  pageAlert: undefined,
+  setPageError: (_: React.ReactNode | undefined) => {},
+  setPageInfo: (_: React.ReactNode | undefined) => {},
+  setPageMuted: (_: React.ReactNode | undefined) => {},
+  setPageSuccess: (_: React.ReactNode | undefined) => {},
+  setPageWarning: (_: React.ReactNode | undefined) => {},
```

Comment:
Should this be `void` instead? The return value is not usable, is it?

```suggestion
  setPageError: (_: React.ReactNode | undefined) => void,
  setPageInfo: (_: React.ReactNode | undefined) => void,
  setPageMuted: (_: React.ReactNode | undefined) => void,
  setPageSuccess: (_: React.ReactNode | undefined) => void,
  setPageWarning: (_: React.ReactNode | undefined) => void
```